packages/govuk-frontend/src/govuk/govuk-frontend-component.mjs

```diff
@@ -8,27 +8,43 @@ import { InitError, SupportError } from './errors/index.mjs'
  *
  * @internal
  * @virtual
+ * @template [T=HTMLElement]
  */
 export class GOVUKFrontendComponent {
+  /**
+   * @protected
+   * @type {T}
+   */
+  $root
+
   /**
    * Constructs a new component, validating that GOV.UK Frontend is supported
    *
    * @internal
-   * @param {Element | null} [$root] - HTML element to use for component
+   * @param {T | null} [$root] - HTML element to use for component
    */
   constructor($root) {
     const childConstructor = /** @type {ChildClassConstructor} */ (
       this.constructor
     )
 
+    if (!($root instanceof HTMLElement)) {
```

Comment:
**issue** We'll need the `HTMLElement` in that check to be provided by child classes (as a static property) so we can ensure that `SkipLink` indeed receives an `HTMLAnchorElement`.

In the current state, if someone sets `data-module="govuk-skip-link"` on a `<div>`, for example, the component would initialise properly as TypeScript doesn't add any checks for when the code runs.